src/React/Basic.js

```diff
@@ -1,39 +1,101 @@
 "use strict";
 
-var React = require('react');
-
-exports.react_ = function(spec) {
-  return React.createClass({
-    getInitialState: function() {
-      return spec.initialState;
-    },
-    componentDidMount: function() {
-      var this_ = this;
-      spec.receiveProps(this.props, this.state, function(newState) {
-        return function() {
-          this_.setState(newState);
-        };
-      });
-    },
-    componentWillReceiveProps: function(newProps) {
-      var this_ = this;
-      spec.receiveProps(newProps, this.state, function(newState) {
-        return function() {
-          this_.setState(newState);
-        };
-      });
-    },
-    render: function() {
-      var this_ = this;
-      return spec.render(this.props, this.state, function(newState) {
-        return function() {
-          this_.setState(newState);
-        };
-      });
-    }
+var React = require("react");
+var Fragment = React.Fragment || "div";
+
+// Class polyfill
+var __extends =
+  (this && this.__extends) ||
+  (function() {
+    var extendStatics =
+      Object.setPrototypeOf ||
+      ({ __proto__: [] } instanceof Array &&
+        function(d, b) {
+          d.__proto__ = b;
+        }) ||
+      function(d, b) {
+        for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
+      };
+    return function(d, b) {
+      extendStatics(d, b);
+      function __() {
+        this.constructor = d;
+      }
+      d.prototype =
+        b === null
+          ? Object.create(b)
+          : ((__.prototype = b.prototype), new __());
+    };
+  })();
+
+var __shallowCopy = function(obj) {
+  if (obj == null) return;
+  var result = {};
+  Object.keys(obj).forEach(function(key) {
+    result[key] = obj[key];
   });
+  return result;
+};
+
+exports.component_ = function(spec) {
+  function Component(props) {
+    this.state = spec.initialState;
+    return this;
+  }
+  __extends(Component, React.Component);
```

Comment:
What happens if you leave this out, out of curiosity?

Reply:
I don't think it will work, but I can try later.  It'd be the same as creating a class to use as a component in ES2015 without adding `extends React.Component`.

Reply:
I tried this, which worked:

```js
Component.prototype = Object.create(React.Component.prototype);
```

Reply:
I almost went that route originally but I was worried there might be some other implementation detail of `class` I wasn't aware of.  I'm fine with using that instead though until we find a problem.